docarray/array/mixins/find.py

```diff
@@ -36,7 +36,7 @@ def find(self: 'T', query: str, **kwargs):
 
     def find(
         self: 'T',
-        query: Union['DocumentArray', 'Document', 'ArrayType'],
+        query: Union['DocumentArray', 'Document', 'ArrayType', Dict],
```

Comment:
I do not think this should be the query, we should at least prepare the interface fo be able to accept both arguments, query and filters

Reply:
I got your point. But we can use another argument, e.g., `pre_filters`. For example, in annlite,
```
docs = DocumentArray(storage='annlite')

docs.find(query, pre_filters={'tags__x': {'$gte': 1.0}})
```
And the query language of `pre_filtering` only depends on the backend.

Reply:
but isn't it weird to see that a `filter` can be put on both sides?

Reply:
@JoanFM I think using `find` would be more reasonable. for your above question, i think this would not be a big deal with well-documented.